src/core/thread/mesh_forwarder.cpp

```diff
@@ -686,14 +686,27 @@ void MeshForwarder::SetRxOnWhenIdle(bool aRxOnWhenIdle)
 {
     Get<Mac::Mac>().SetRxOnWhenIdle(aRxOnWhenIdle);
 
+#if OPENTHREAD_CONFIG_WAKEUP_END_DEVICE_ENABLE
+    // Data polls not allowed in enhanced CSL mode
```

Comment:
Now that the enhanced CSL is not done with the sleepy parent, is blocking data polls relevant anymore? This can be cleaned up in a followup though.

Reply:
Yeah, to be decided, let's keep it for now.